crates/gen/src/lib.rs

```diff
@@ -25,9 +25,16 @@ fn format_ident(name: &str) -> proc_macro2::Ident {
     if name == "Self" {
         quote::format_ident!("{}_", name)
     } else {
-        match syn::parse_str::<proc_macro2::Ident>(name) {
-            Ok(i) => i,
-            Err(_) => quote::format_ident!("r#{}", name),
+        // keywords list based on https://doc.rust-lang.org/reference/keywords.html
+        match name {
```

Comment:
I'm just not sure whether this is worthwhile over just escaping all names. If its about code completion, we should fix rust-analyzer to demangle the name. At the very least we should check what if any impact this has on code gen time.

Reply:
This list is stagnate so it's not a maintenance burden really. We should check performance though. This is actually what we were doing before just now checking for a few additional keywords.

Reply:
Tested this and it doesn't appear to make a measurable difference. 👍